db/migrations/20210122160549_update_get_flop_query.sql

```diff
@@ -0,0 +1,187 @@
+-- +goose Up
+-- add a flop_address attribute to the flop_bid_snapshot type
+ALTER TYPE api.flop_bid_snapshot ADD ATTRIBUTE flop_address text;
+
+-- add a new function that works the same as get_flop but takes into account a specific flop_address
+-- this is required because there are now more than one flop contract
+CREATE OR REPLACE FUNCTION api.get_flop_with_address(bid_id NUMERIC, flop_address TEXT,
+                                                     block_height BIGINT DEFAULT api.max_block())
+    RETURNS api.flop_bid_snapshot
+AS
+$$
+WITH address_id AS (SELECT id FROM public.addresses WHERE address = get_flop_with_address.flop_address),
+     storage_values AS (
+         SELECT flop.bid_id,
+                guy,
+                tic,
+                "end",
+                lot,
+                bid,
+                created,
+                updated
+         FROM maker.flop
+         WHERE flop.bid_id = get_flop_with_address.bid_id
+           AND flop.address_id = (SELECT * FROM address_id)
+           AND block_number <= block_height
+         ORDER BY block_number DESC
+         LIMIT 1
```

Comment:
Isn't it still dangerous to use LIMIT 1 here?

Reply:
I don't think so, because any records that share the same bid ID and address should be at different blocks - so this should just give us the most recent snapshot of a given flop auction. Let me know if there's a scenario I'm missing

Reply:
The order is probably fixing it, I think you're right.